services/registration/src/routes/application.ts

```diff
@@ -9,17 +11,58 @@ export const applicationRouter = express.Router();
 
 applicationRouter.route("/").get(
   asyncHandler(async (req, res) => {
-    const applications = await ApplicationModel.find({});
+    const filter: FilterQuery<Application> = {};
 
-    return res.send(applications);
+    if (req.query.hexathon) {
+      filter.hexathon = req.query.hexathon;
+    }
+
+    if (req.query.userId) {
+      filter.userId = req.query.userId;
+    }
+
+    const applications = await ApplicationModel.find(filter);
+    const userIdArray: string[] = [];
+    for (const application of applications) {
+      userIdArray.push(application.userId);
+    }
+
+    const userInfo = await apiCall(
+      Service.USERS,
+      { method: "POST", url: `users/actions/retrieve`, data: userIdArray },
+      req
+    );
+
+    const finalApps = [];
+
+    for (const info of userInfo) {
+      for (const application of applications) {
+        const totApp = {
+          ...application,
+          userInfo: info,
+        };
+        finalApps.push(totApp);
+      }
+    }
+
+    return res.send(finalApps);
   })
 );
 
 applicationRouter.route("/:id").get(
   asyncHandler(async (req, res) => {
     const application = await ApplicationModel.findById(req.params.id);
+    let userInfo = {};
+    if (application) {
+      userInfo = apiCall(Service.USERS, { method: "GET", url: `users/${application.userId}` }, req);
+    }
+
+    const finalApp = {
+      ...application,
+      user: userInfo,
```

Comment:
Between here and the other endpoint, can you keep it consistent to userInfo as the field key?